CONTRIBUTING.md

```diff
@@ -0,0 +1,168 @@
+
+If you need any help or mentoring getting started, understanding the codebase, or making a PR (or anything else really), please ask in `#sig-dashboard` channel on [Slack](https://join.slack.com/t/tidbcommunity/shared_invite/enQtNzc0MzI4ODExMDc4LWYwYmIzMjZkYzJiNDUxMmZlN2FiMGJkZjAyMzQ5NGU0NGY0NzI3NTYwMjAyNGQ1N2I2ZjAxNzc1OGUwYWM0NzE).
+
+## Building and setting up a development workspace
+
+TiDB Dashboard is a web based UI for TiDB clusters. It is integrated into [PD](https://github.com/pingcap/pd) by default. It can also be deployed as a standalone server. The back-end is written in [Golang](https://golang.org/) with [Gin web framework](https://github.com/gin-gonic/gin) and the front-end is written in [TypeScript](https://www.typescriptlang.org/) with [React](https://github.com/facebook/react). [swaggo](https://github.com/swaggo/swag) and [openapi-generator](https://github.com/OpenAPITools/openapi-generator) are used for automatically generating API documents and API client.
+
+The following steps are describing how to develop TiDB Dashboard by running a self-built TiDB Dashboard server (i.e. standalone) along with a separated TiDB cluster.
+
+### Prerequisites
+
+To build TiDB Dashboard you'll need to at least have the following installed:
+
+- `git` - Version control
+- `make` - Build tool (run common workflows)
+- [`Golang`](https://golang.org/) - Golang (require 1.13+)
+- [`Node.js`](https://nodejs.org/) - Javascript runtime (require 12+)
+- [`Yarn`](https://classic.yarnpkg.com/en/docs/install) - Javascript dependency management (require 1.21+)
+- [`Java`](https://www.java.com/ES/download/) - depended by [openapi-generator](https://github.com/OpenAPITools/openapi-generator) (require 8+)
+
+### Getting the repository
+
+```bash
+git clone https://github.com/pingcap-incubator/tidb-dashboard.git
+cd tidb-dashboard
+# Future instructions assume you are in this directory
+```
+
+Optional: set Taobao's npm mirror registry to speed up package downloading if needed
+
+```bash
+npm config set registry https://registry.npm.taobao.org
+```
+
+### Building and running
+
+At this point, you can build and run TiDB Dashboard. 
+
+> Note: If you want to debug TiDB Dashboard, it needs a running TiDB cluster as target, see [how to start a local TiDB cluster](#starting-a-tidb-cluster).
+
+Build and run back-end server:
+
+```bash
+make
+make run
+# Now tidb-dashboard server is listen on 127.0.0.1:12333
+```
+
+For front-end, you should build API client and start a React development server:
+
+```bash
+cd ui
+npm run build_api_client
+npm run start
+# Now tidb-dashboard UI is available on 127.0.0.1:3000
+```
+
+When you're ready to test out your changes, use the `dev` task. It will lint your code and verify the UI building.
+
+```bash
+make dev
```

Comment:
this part only applies to back-end develop. Maybe better to put it before `For front-end, ...`

Reply:
As we will add `js linter` in the future, I think we can keep content here and add front-end check command here in the future.